python/ray/serve/_private/deployment_state.py

```diff
@@ -2779,6 +2791,26 @@ def update(self) -> bool:
 
         return any_recovering
 
+    def _handle_scheduling_request_failures(
+        self,
+        deployment_id: DeploymentID,
+        scheduling_requests: List[ReplicaSchedulingRequest],
+    ):
+        """Updates internal datastructures when replicas fail to be scheduled."""
+        failed_replicas: List[ReplicaID] = []
+        for scheduling_request in scheduling_requests:
+            if scheduling_request.scheduling_failed:
+                failed_replicas.append(scheduling_request.replica_id)
+                error_msg = "Replica scheduling failed."
+                if scheduling_request.scheduling_failed_reason:
+                    error_msg += f" {scheduling_request.scheduling_failed_reason}"
```

Comment:
nit: no need f-string here.

Reply:
The alternative seems a bit messier to me:

```
error_msg = error_msg + " " + scheduling_request.scheduling_failed_reason
```

so I'll keep as is.